src/sync.js

```diff
@@ -504,6 +504,38 @@ SyncApi.prototype.retryImmediately = function() {
     return true;
 };
 
+SyncApi.prototype._syncFromCache = async function(savedSync) {
+    debuglog("sync(): not doing HTTP hit, instead returning stored /sync data");
+
+    const oldSyncToken = this.client.store.getSyncToken();
+    const nextSyncToken = savedSync.nextBatch;
+
+    this.client.store.setSyncToken(nextSyncToken);
```

Comment:
is this really right? it looks like we're going to update the stored sync token each time we start up, even if we're offline, which can't be right?

Reply:
If we're offline, it'll always be the same - the `nextBatch` of the cached sync. When we start syncing, we use that token as the starting point. 

This was happening before. Am I missing something here?

Reply:
so you're saying that `oldSyncToken == nextSyncToken` in this case, so this is a no-op?

Reply:
I agree it does look like it was happening before, but I remain confused about how it actually works

Reply:
>so you're saying that oldSyncToken == nextSyncToken in this case, so this is a no-op?